src/ui/pages/SurveyUnitListPage.tsx

```diff
@@ -0,0 +1,137 @@
+  const notifier = useNotifier();
+  const orchestratorUrl = process.env.REACT_APP_ORCHESTRATOR_URL;
+  const { questionnaireId, modeName } = useParams<SurveyUnitParams>();
+
+  useEffect(() => {
+    // react router doesn't type, but a check has already been done in Application.tsx ... (\\d+))
+    if (!(questionnaireId && modeName)) {
+      return;
+    }
+    const questionnaireIdNumber = parseInt(questionnaireId);
+    loadSurveyUnitsData(questionnaireIdNumber, modeName);
+
+    questionnaireUseCase
+      .getQuestionnaire(questionnaireIdNumber)
+      .then((questionnaireData) => {
+        setQuestionnaire(questionnaireData);
+        setQuestionnaireLoading(false);
+      });
+  }, [questionnaireId, modeName]);
+
+  const loadSurveyUnitsData = (questionnaireId: number, mode: string): void => {
+    setSurveyUnitsLoading(true);
+    surveyUnitUseCase
+      .getSurveyUnitsData(questionnaireId, mode)
+      .then((surveyUnitsData) => {
+        setSurveyUnitsData(surveyUnitsData);
+        setSurveyUnitsLoading(false);
+      })
+      .catch((err) => {
+        notifier.error(intl.formatMessage({ id: "error_request_failed" }));
+        console.log(err);
+      });
+  };
+
+  return (
+    <Grid component="main" container justifyContent="center">
+      <Grid item xs={12} md={6}>
+        <Block>
+          <Loader isLoading={isSurveyUnitsLoading && isQuestionnaireLoading}>
+            <Title>
+              {intl.formatMessage({ id: "survey_unit_list_label" })}
+            </Title>
+
+            <Subtitle>
+              <>
+                {questionnaire?.label}
+                <br />
+                {intl.formatMessage(
+                  { id: "survey_unit_mode_label" },
+                  { modeName: modeName }
+                )}
+              </>
+            </Subtitle>
+
+            <TableContainer component={Paper}>
+              <Table aria-label="surveyUnit table">
+                <TableHead>
+                  <TableRow>
+                    <TableCell>
+                      {intl.formatMessage({
+                        id: "survey_unit_id",
+                      })}
+                    </TableCell>
+                    <TableCell align="center">
+                      {intl.formatMessage({
+                        id: "survey_unit_list_actions",
+                      })}
+                    </TableCell>
+                  </TableRow>
+                </TableHead>
+                {
+                  <TableBody>
+                    {questionnaire &&
+                      surveyUnitsData?.surveyUnits?.map((surveyUnit) => (
+                        <TableRow key={surveyUnit.id}>
+                          <TableCell component="th" scope="row">
+                            {surveyUnit.displayableId}
+                          </TableCell>
+                          <TableCell align="center">
+                            <a
+                              target="_blank"
```

Comment:
the target blank is not very accessible or you have to put additional information.